js/src/button.js

```diff
@@ -140,15 +143,19 @@ class Button {
 
 $(document)
   .on(Event.CLICK_DATA_API, Selector.DATA_TOGGLE_CARROT, (event) => {
-    event.preventDefault()
-
     let button = event.target
 
     if (!$(button).hasClass(ClassName.BUTTON)) {
-      button = $(button).closest(Selector.BUTTON)
+      button = $(button).closest(Selector.BUTTON)[0]
     }
 
-    Button._jQueryInterface.call($(button), 'toggle')
+    if (!button || button.hasAttribute('disabled') || button.classList.contains('disabled')) {
+      event.preventDefault() // work around Firefox bug #1540995
+    } else if (button.querySelector(Selector.INPUT) && (button.querySelector(Selector.INPUT).hasAttribute('disabled') || button.querySelector(Selector.INPUT).classList.contains('disabled'))) {
```

Comment:
Here you call 3 times `querySelector` please store the returned value.

Something like:
```js
if (!button || button.hasAttribute('disabled') || button.classList.contains('disabled')) {
  event.preventDefault() // work around Firefox bug #1540995
} else {
  const inputBtn = button.querySelector(Selector.INPUT)

  if (inputBtn && (inputBtn.hasAttribute('disabled') || inputBtn.classList.contains('disabled')) {
    event.preventDefault() // work around Firefox bug #1540995
    return
  }

  Button._jQueryInterface.call($(button), 'toggle')
}